live-ingester/src/test/java/org/alfresco/hxi_connector/live_ingester/adapters/storage/endpoint/PreSignedUrlRequesterIntegrationTest.java

```diff
@@ -0,0 +1,120 @@
+import org.apache.camel.spring.boot.CamelAutoConfiguration;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.test.context.ActiveProfiles;
+import org.springframework.test.context.DynamicPropertyRegistry;
+import org.springframework.test.context.DynamicPropertySource;
+import org.testcontainers.junit.jupiter.Container;
+import org.testcontainers.junit.jupiter.Testcontainers;
+import org.testcontainers.utility.DockerImageName;
+import org.wiremock.integrations.testcontainers.WireMockContainer;
+
+import org.alfresco.hxi_connector.live_ingester.adapters.config.MappingConfig;
+import org.alfresco.hxi_connector.live_ingester.adapters.storage.StorageLocationRequest;
+import org.alfresco.hxi_connector.live_ingester.adapters.storage.StorageLocationRequester;
+import org.alfresco.hxi_connector.live_ingester.util.DockerTags;
+
+@SpringBootTest(classes = {
+        CamelAutoConfiguration.class,
+        MappingConfig.class,
+        PreSignedUrlRequester.class})
+@ActiveProfiles({"test"})
+@Testcontainers
+class PreSignedUrlRequesterIntegrationTest
+{
+    private static final String WIREMOCK_IMAGE = "wiremock/wiremock";
+    private static final String WIREMOCK_TAG = DockerTags.getOrDefault("wiremock.tag", "3.3.1");
+
+    private static final String HX_INSIGHT_PRE_SIGNED_URL_PATH = "/pre-signed-url";
+    private static final String HX_INSIGHT_TEST_USERNAME = "mock";
+    private static final String HX_INSIGHT_TEST_PASSWORD = "pass";
+    private static final String CAMEL_ENDPOINT_PATTERN = "%s%s?httpMethod=POST&authMethod=Basic&authUsername=%s&authPassword=%s&authenticationPreemptive=true";
+    private static final String FILE_CONTENT_TYPE = "plain/text";
+    private static final String PRE_SIGNED_URL = "http://s3-storage-location";
+    private static final String HX_INSIGHT_RESPONSE_BODY = String.format("{\"%s\": \"%s\"}", STORAGE_LOCATION_PROPERTY, PRE_SIGNED_URL);
+    private static final int HX_INSIGHT_RESPONSE_CODE = 201;
+
+    @Container
+    @SuppressWarnings("PMD.FieldNamingConventions")
+    static final WireMockContainer wireMockServer = new WireMockContainer(DockerImageName.parse(WIREMOCK_IMAGE).withTag(WIREMOCK_TAG));
+
+    @Autowired
+    StorageLocationRequester locationRequester;
+
+    @BeforeAll
+    static void beforeAll()
+    {
+        WireMock.configureFor(wireMockServer.getHost(), wireMockServer.getPort());
+    }
+
+    @Test
+    void testRequestStorageLocation()
+    {
+        // given
+        String nodeId = "some-node-ref";
+        givenThat(post(HX_INSIGHT_PRE_SIGNED_URL_PATH)
+                .withBasicAuth(HX_INSIGHT_TEST_USERNAME, HX_INSIGHT_TEST_PASSWORD)
+                .willReturn(aResponse()
+                        .withStatus(HX_INSIGHT_RESPONSE_CODE)
+                        .withBody(HX_INSIGHT_RESPONSE_BODY)));
```

Comment:
I think this would be clearer if the expected return value `PRE_SIGNED_URL` was explicitly mentioned in the given step. E.g. Use a local variable for `HX_INSIGHT_RESPONSE_BODY`.

Reply:
Fixed according to your suggestion